x/txfees/keeper/feedecorator.go

```diff
@@ -57,6 +58,12 @@ func (mfd MempoolFeeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate b
 		return ctx, types.ErrTooManyFeeCoins
 	}
 
+	// TODO: Is there a better way to do this?
+	// I want ctx.IsDeliverTx() but that doesn't exist.
+	if !ctx.IsCheckTx() && !ctx.IsReCheckTx() {
```

Comment:
@ValarDragon why not just implement this as a custom prepare proposal handler?

Reply:
Osmosis is still on its custom fork of the Cosmos SDK.  In the next upgrade, we will be finishing our unfork, which will give us access to ABCI++

https://github.com/osmosis-labs/osmosis/pull/6413

Reply:
ah my bad!